foosballv2s/MovementDetector.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Emgu.CV;
+using Emgu.CV.CvEnum;
+using Emgu.CV.Structure;
+using System;
+using System.Drawing;
+
+namespace foosballv2s
+{
+    class MovementDetector
+    {
+        private IVideo video;
+
+        public MovementDetector(IVideo video)
+        {
+            this.video = video;
+        }
+       
+        // Steps:
+        // 1) A frame is retrieved from a video
+        // 2) The frame is converted from RGB to HSV color scheme (for more accurate detection)
+        // 3) All unneeded colors are filtered out, only pixels with given color are left
+        // 4) After this step the picture is in grayscale where the ball appears white and all other
+        //    pixels are black
+        // 5) Circles are detected in the grayscale image
+        // 6) The proccessed frame with circles drawn on it is shown in a window
+        public bool DetectBall()
+        {
+            // Currently the color is hardcoded here. Needs to be detected somewhere else
+            // and passed here as parameters
+            int t1min = 170, t1max = 4, t2min = 120, t2max = 255, t3min = 120, t3max = 255; // color in HSV
```

Comment:
As I understand it, this part will be left out in the main project as we will opt for picture color recognition. But the question remains how large will we leave the opening for color changes.

Reply:
Probably we'll see what range of colors should be picked for best results as we continue to develop further.